dali/dfuXRefLib/dfuxreflib.cpp

```diff
@@ -757,6 +760,42 @@ struct CLogicalNameEntry: public CInterface
         return false;
     }
 
+    RemoteFilename &constructPartFilename(unsigned partNo, unsigned copy, RemoteFilename &rfn, IPropertyTree &file)
+    {
+        partNo--;
+        StringBuffer partName;
+        if (pmask.isEmpty())
+        {
+            pmask.set("!ERROR!._$P$_of_$N$");
+            IERRLOG("No partmask for CLogicalNameEntry::constructPartFilename");
+        }
+        expandMask(partName, pmask, partNo, max);
+
+        // Get stripeNum from storage plane
+        Owned<IStoragePlane> plane = getDataStoragePlane(grpname, true);
+        const char * prefix = plane->queryPrefix();
+        unsigned stripeNum = calcStripeNumber(partNo+1, lfnHash, plane->numDevices());
+
+        // Get dir-per-part # from file metadata or storage plan info
+        FileDescriptorFlags flags = static_cast<FileDescriptorFlags>(file.getPropInt("Attr/@flags"));
```

Comment:
@jakesmith I added file to the parameters for constructPartFilename, but I think that may be incorrect since it uses file in the CLogicalNameEntry constructor. Which is considered best practice, initializing CLogicalNameEntry with the dir-per-part/stripe-number and adding additional members to the class or passing file into constructPartFilename and getting the information right when we need it?